src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -768,6 +769,33 @@ AcquireRebalanceColocationLock(Oid relationId, const char *operationName)
 }
 
 
+/*
+ * AcquireRebalanceOperationLock does not allow concurrent rebalance
+ * operations.
+ */
+static void
+AcquireRebalanceOperationLock(const char *operationName)
+{
+	LOCKTAG tag;
+	const bool sessionLock = false;
+	const bool dontWait = true;
+
+	SET_LOCKTAG_CITUS_OPERATION(tag, CITUS_REBALANCE_OPERATION);
```

Comment:
Is there a possible (albeit rare) race condition where the lock could be acquired by two different operations on different nodes?